templates/.github/workflows/go-test.yml

```diff
@@ -6,12 +6,15 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ "ubuntu", "windows", "macos" ]
         go: [ "1.16.x", "1.17.x" ]
+        os:
+          - { name: "ubuntu", shell: "bash" }
+          - { name: "macos", shell: "bash" }
+          - { name: "windows", shell: "msys2 {0}" }
```

Comment:
I wonder if we could derive the shell by default from the OS name, to reduce verbosity.

Reply:
Not without a separate step I think.

Reply:
NOTE: this is a breaking change if someone started using `os.name` in their custom `go-test-setup/action.yml`. I don't expect there are many (if any) places like this so I can just keep an eye out for failing sync PRs once we propagate this change throughout the repos and fix where needed.